runelite-client/src/main/java/net/runelite/client/plugins/barbarianassault/BarbarianAssaultPlugin.java

```diff
@@ -94,76 +107,109 @@ protected void startUp() throws Exception
 	protected void shutDown() throws Exception
 	{
 		overlayManager.remove(overlay);
+		gameTime = null;
+		currentWave = "1";
+		inGameBit = 0;
 	}
 
 	@Subscribe
-	public void onGameTick(GameTick event)
+	public void onWidgetLoaded(WidgetLoaded event)
 	{
-		if (client.getVar(Varbits.IN_GAME_BA) == 1 &&
-				overlay.getCurrentRound() == null &&
-				client.getLocalPlayer() != null)
+		if (event.getGroupId() == WidgetID.BA_REWARD_GROUP_ID)
 		{
-			switch (client.getLocalPlayer().getPlayerComposition().getEquipmentId(KitType.CAPE))
+			Widget rewardWidget = client.getWidget(WidgetInfo.BA_REWARD_TEXT);
+
+			if (config.waveTimes() && rewardWidget != null && rewardWidget.getText().contains("<br>5") && gameTime != null)
```

Comment:
I'm using the "<br>5" string because it ONLY occurs in the end game reward screen. So when this string is detected in the BA_REWARD_TEXT Widget, it triggers the GameTimer to stop.

Reply:
Yeah that is why you should document it either by named constant or comment

Reply:
Okay I will do that 👍